semantic_release/history/__init__.py

```diff
@@ -233,7 +226,7 @@ def get_current_release_version_by_tag() -> str:
 
     :return: A string with the version number or 0.0.0 on failure.
     """
-    version = get_last_version(pattern=get_release_version_pattern())
+    version = get_last_version(pattern=get_tag_pattern())
```

Comment:
I don't really think there's a difference between current and release patterns when using tags, so these functions are essentially duplicates when `tag_source == "tag*"`